.github/workflows/enzyme-rust.yml

```diff
@@ -0,0 +1,62 @@
+name: Enzyme Rust CI
+
+on: 
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  merge_group:
+
+jobs:
+  rust-autodiff:
+    name: Rust Autodiff Tests LLVM ${{ matrix.llvm }}
+    runs-on: [self-hosted, Linux, X64, 32-core]
```

Comment:
this is not the correct setup for the runner, see https://github.com/EnzymeAD/Enzyme/blob/main/.github/workflows/enzyme-mlir.yml for an example

Reply:
it needs to run on `linux-x86-n2-32`